dvc/repo/get.py

```diff
@@ -20,26 +21,25 @@ def get(url, path, out=None, rev=None):
     tmp_dir = os.path.join(dpath, "." + str(shortuuid.uuid()))
     erepo = ExternalRepo(tmp_dir, url=url, rev=rev)
     try:
-        erepo.install()
-        # Try any links possible to avoid data duplication.
-        #
-        # Not using symlink, because we need to remove cache after we are
-        # done, and to make that work we would have to copy data over
-        # anyway before removing the cache, so we might just copy it
-        # right away.
-        #
-        # Also, we can't use theoretical "move" link type here, because
-        # the same cache file might be used a few times in a directory.
-        erepo.repo.config.set(
-            Config.SECTION_CACHE,
-            Config.SECTION_CACHE_TYPE,
-            "reflink,hardlink,copy",
-        )
-        src = os.path.join(erepo.path, urlparse(path).path.lstrip("/"))
-        o, = erepo.repo.find_outs_by_path(src)
-        erepo.repo.fetch(o.stage.path)
-        o.path_info = PathInfo(os.path.abspath(out))
-        with o.repo.state:
-            o.checkout()
+        with erepo:
+            # Try any links possible to avoid data duplication.
+            #
+            # Not using symlink, because we need to remove cache after we are
+            # done, and to make that work we would have to copy data over
+            # anyway before removing the cache, so we might just copy it
+            # right away.
+            #
+            # Also, we can't use theoretical "move" link type here, because
+            # the same cache file might be used a few times in a directory.
+            erepo.repo.config.set(
+                Config.SECTION_CACHE,
+                Config.SECTION_CACHE_TYPE,
+                "reflink,hardlink,copy",
+            )
+            o = erepo.get_out(path)
+            erepo.repo.fetch(o.stage.path)
+            o.path_info = PathInfo(os.path.abspath(out))
+            with o.repo.state:
+                o.checkout()
```

Comment:
why do we need to do checkout? why fetch is not enough?

Reply:
Fetch is only downloading the cache, but the checkout is the one that is creating a workspace link to the cache.

Reply:
my bad, didn't notice that it's get